### DISCUSSION TITLE
unable to submit pull request because I'm not a collaborator?

### DISCUSSION THREAD

**Discussion by CrashDavis010** [community/community#121562] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

Sorry, newbie/dummy here.  I'm trying to validate a token I'm in, but have never submitted before.  I have the code, but when I try to create a new pull request, regardless of which base and compare I choose, I am told I'm not a collaborator so I can't submit.  Please help.  Thank you.

**Comment by salaamdev** [comment]
Hey there! The reason you can't create a pull request directly is that the repo you're trying to contribute to likely isn't your own. For external contributions, you'll need to follow the "fork and pull request" workflow:

1. Fork the repo to your account.
2. Clone your fork locally.
3. Create a new branch for your changes.
4. Make your desired changes.
5. Commit and push to your fork.
6. Create a pull request from your fork to the main repo.

**Comment by gurjeetsinghvirdee** [comment, marked as answer]
Hey @CrashDavis010,

I think you are referring to this repository: https://github.com/CrashDavis010/token-list, which you have forked.

Before making changes to this, first, connect your **Git** to **GitHub**:
- First, install **Git** on your system: https://git-scm.com/.
- After this, open a `Git Bash Terminal`.

Run these commands:

1) Set a Git username:
`git config --global user.name CrashDavis010`

2) Confirm the username:
`git config --global user.name`

Setting your commit email address on GitHub:

1) Set a Git Email:
`git config --global user.email "YOUR_EMAIL"` (open settings > Emails > Add Email > Verify your email > Use Github no reply email)
It will look like this `example@users.noreply.github.com`.

2) Confirm Email:
`git config --global user.email`

Now, here are a few steps to make a **First Pull Request**:

- Fork the repo as you have already done.
- Now clone the forked repo to your system. Click the green `Code` button; it will give you a link like shown here:
![image](https://github.com/community/community/assets/73753957/67015d23-62d3-4c72-8cf1-27bb69293e5b).
- Use this command to clone your forked repo: 
`git clone https://github.com/CrashDavis010/token-list.git`.

It will take some time to clone, around 2-3 minutes.
- Then, create a new branch before making any changes on it. Creating a new branch is a good practice in Open Source:
`git checkout -b branch-name`.

- Start your changes. When you're ready, run these commands to push it to your forked repo:

- `git add .` (It will add all changes).
- `git commit -m your-commit-message`.
- `git push -u origin branch-name`.

After this, come back to your forked repo. The changes you made will reflect on it, and there will be a new option: `Create a Pull Request` button. Now you can easily make a pull request.

**Comment by gurjeetsinghvirdee** [comment]
If you still feel confused you can go through this: https://www.youtube.com/watch?v=qYoc07Da6kg

Happy Coding!!
